maestro-orchestra/src/main/java/maestro/orchestra/Orchestra.kt

```diff
@@ -579,16 +585,26 @@ class Orchestra(
         // filter out DefineVariablesCommand to not execute it twice
         val filteredCommands = commands.filter { it.asCommand() !is DefineVariablesCommand }
 
-        subflowConfig?.onFlowStart?.commands?.let {
-            executeSubflowCommands(it, config)
-        }
-
+        var exception: Throwable? = null
+        var flowSuccess = false
         try {
-            return executeSubflowCommands(filteredCommands, config)
-        } finally {
-            subflowConfig?.onFlowComplete?.commands?.let {
+            val onStartSuccess = subflowConfig?.onFlowStart?.commands?.let {
                 executeSubflowCommands(it, config)
+            } ?: true
+
+            if (onStartSuccess) {
+                flowSuccess = executeSubflowCommands(filteredCommands, config)
             }
+        } catch (e: Throwable) {
+            exception = e
+        } finally {
+            val onCompleteSuccess = subflowConfig?.onFlowComplete?.commands?.let {
+                executeSubflowCommands(it, config)
+            } ?: true
+
+            exception?.let { throw it }
+
+            return onCompleteSuccess && flowSuccess
```

Comment:
This bit confuses me, why not:
```
   return onComplete && success
} finally {
  onFlowComplete.execute()
}
```

Reply:
the thing here is we also need to know `onFlowComplete.execute()` result before returning.

in case `onFlowComplete.execute()` returns false we mark flow as failed as documented. 

also `onFlowComplete.execute()` needs to be executed nevermind the result of the main flow execution